homeassistant/components/opnsense/config_flow.py

```diff
@@ -0,0 +1,161 @@
+"""Config flow for OPNsense."""
+
+import logging
+from typing import Any
+
+from pyopnsense import diagnostics
+from pyopnsense.exceptions import APIException
+from requests.exceptions import ConnectionError as requestsConnectionError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_HOST,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_URL,
+    CONF_VERIFY_SSL,
+)
+
+from .const import (
+    CONF_API_BASE_URL,
+    CONF_API_SECRET,
+    CONF_API_VERIFY_CERT,
+    CONF_TRACKER_INTERFACES,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OPNsenseConfigFlow(ConfigFlow, domain=DOMAIN):
+    """OPNsense config flow."""
+
+    VERSION = 1
+
+    async def _show_setup_form(
+        self,
+        user_input: (dict[Any, Any] | None) = None,
+        errors: (dict[Any, Any] | None) = None,
+    ) -> ConfigFlowResult:
+        """Show the setup form to the user."""
+        if user_input is None:
+            user_input = {}
+
+        description_placeholders = {}
+        description_placeholders["doc_url"] = (
+            "https://www.home-assistant.io/integrations/opnsense/"
+        )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+                    vol.Required(
+                        CONF_PORT, default=user_input.get(CONF_PORT, 80)
+                    ): vol.Coerce(int),
+                    vol.Required(
+                        CONF_API_KEY, default=user_input.get(CONF_API_KEY, "")
+                    ): str,
+                    vol.Required(
+                        CONF_API_SECRET, default=user_input.get(CONF_API_SECRET, "")
+                    ): str,
+                    vol.Required(
+                        CONF_SSL, default=user_input.get(CONF_SSL, False)
+                    ): bool,
+                    vol.Required(
+                        CONF_VERIFY_SSL, default=user_input.get(CONF_VERIFY_SSL, False)
+                    ): bool,
+                    vol.Optional(CONF_TRACKER_INTERFACES): str,
```

Comment:
The reason you probably didn't get the dropdown working is that there isn't a way to do it dynamically. What would most users do? Enable all? Or just a few? If the average user has to edit this, I would recommend a separate step, otherwise we can just enable all or disable all and give them an options flow to change it